### PR TITLE
Add a 'target' field for symlinkattributes

### PR DESCRIPTION
**Pull request by hishamhm**
It returns the resolved path of the symlink.

Since this adds functionality, this should be merged in the 1.7 branch. PRing for review.

### CODE REVIEW

**Comment by hishamhm** [comment]
(I don't know why these other commits 6e0dc7b and 7dae11c ended up in this PR too, but I guess they should be in `1.7` too)

File: src/lfs.c [hunk 1]
```diff
@@ -850,10 +850,73 @@ static int file_info (lua_State *L) {
...
 
 /*
+** Push the symlink target to the top of the stack.
+** Assumes the file name is at position 1 of the stack.
+** Returns 1 if successful (with the target on top of the stack),
+** 0 on failure (with stack unchanged, and errno set).
+*/
+static int push_link_target(lua_State *L) {
+#ifdef _WIN32
+        errno = ENOSYS;
```
Comment (daurnimator) [inline src/lfs.c]: Why not implement on windows?

Comment (hishamhm) [inline src/lfs.c]: Be my guest! :)

Comment (daurnimator) [inline src/lfs.c]: So that implies it was just a lack of time rather than a technical obstacle?

Comment (hishamhm) [inline src/lfs.c]: ...and interest — I usually don't do Windows coding for free :)

**Comment by hishamhm** [comment]
For cross-referencing: I noticed that this closes #43.

**Comment by mpeterv** [comment]
> (I don't know why these other commits 6e0dc7b and 7dae11c ended up in this PR too, but I guess they should be in 1.7 too)

Probably because I haven't merged `master` into `1.7` yet and this PR's branch was created from `master`.

**Comment by mpeterv** [comment]
Not sure what is going on with LuaJIT failures, see mpeterv/hererocks#27.

**Comment by mpeterv** [comment]
@hishamhm I don't like that to getting size for `readlink` requires indexing a table that was just built. Maybe could return stat struct (using pointer argument) to make post-processing easier?

**Comment by mpeterv** [comment]
Alternatively, it should be possible to use existing mechanism for declaring functions pushing fields. But then there has to be a way to use different field lists for `stat` and `lstat`, and also these pushing functions should be able to signal failure.

**Comment by n1tehawk** [comment]
I'd like to suggest a different approach for determining the buffer size:

Start with a reasonable initial value (say 256 or 512), and check the `readlink()` return value against it. If that suggests that the buffer may have been filled completely, `realloc()` it to a bigger size and retry. See https://github.com/n1tehawk/luafilesystem/commit/3497406ff94f742c890e0a5ad79bd78048c36c55#diff-f6b7bcefa5f97fba6fa9e82b43357724R865 ...

If you wish, I'll create a pull request for this.
Regards, NiteHawk

**Comment by hishamhm** [comment]
@n1tehawk yes, this PR would be welcome!

**Comment by n1tehawk** [comment]
See #78.

Compared to my previous suggestion, I have added proper `readlink()` error handling. Also: The size restriction was too conservative/narrow without need - since we'll only write `target[tsize]` if (tsize < size), it's safe to use `size` for the readlink() call directly (instead of `size - 1`).

Regards, NiteHawk

**Comment by n1tehawk** [comment]
This PR was superseded by merging #78, and should be closed.

Regards, NiteHawk

**Pull request closed without merging**